tests/suites/test_suite_rsa.function

```diff
@@ -49,7 +50,14 @@ void mbedtls_rsa_pkcs1_sign( char *message_hex_string, int padding_mode, int dig
 
     TEST_ASSERT( mbedtls_rsa_import( &ctx, &N, &P, &Q, NULL, &E ) == 0 );
     TEST_ASSERT( mbedtls_rsa_get_len( &ctx ) == (size_t) ( mod / 8 ) );
-    TEST_ASSERT( mbedtls_rsa_complete( &ctx ) == 0 );
+
+    res = mbedtls_rsa_complete( &ctx );
+#if defined(MBEDTLS_RSA_ALT)
+    if( res == MBEDTLS_ERR_RSA_UNSUPPORTED_OPERATION )
```

Comment:
If `mbedtls_rsa_complete` returns `MBEDTLS_ERR_RSA_UNSUPPORTED_OPERATION`, then the test passes, but the objective of the test (calling `mbedtls_rsa_pkcs1_sign`) is not performed. This could easily hide failures behind a misleading test report: the test report could say the many RSA signature tests passed when in fact no RSA signature operation was ever done and RSA signature could be completely broken.

Reply:
@gilles-peskine-arm Do you have a concrete idea how we could improve on this? The only alternative I see is to designate a few test cases with "standard" parameters that every alternative implementation must accept (e.g. all of N,E or N,D,E,P,Q, plus the parameters have standard sizes and values), but I don't think we want to put that limitation on alternative implementations (I don't think we can single out a single bit length that should be supported by every alternative implementation). This could only be resolved by additional flags indicating, for example, the supported RSA bit lengths.

Reply:
Is it possible for a test to skip itself instead of passing (or failing)? This will communicate that the test was not run due to a lack of hardware support.

Additional, if, for example, the hardware only supports 1024-, 2048-, 4096-bit keys, we should ensure we have enough test cases to ensure that we can actually test something on the hardware by testing with those supported key sizes.

Reply:
@gilles-peskine-arm @mazimkhan Is it possible to skip a test after it has been started, so that at least in the test output we'd see lots of `SKIPPED` instead of `PASSED` in case of alternative implementations that refute the majority of our tests?

Reply:
At the moment, once the test case runs, it's either PASS or FAIL (or crash). There may be a way to hack this in, but note that we'd want it to work both in the current framework and with the upcoming rewrite in #930 .

Reply:
Currently, tests set modify a global variable `test_info`, in particular setting a failure indicator field `test_info.failed` when a `TEST_ASSERT` fails. The orchestrating code builds on this variable to report the test as `PASS` or `FAIL`. I seems that this could easily be made more fine grained by having another value for `test_info.failed` which indicates that the test has been skipped.

Reply:
Well, if it isn't possible now, we can plan to add the capability (if we want it) to address the concerns of @gilles-peskine-arm. The same goes for the additional test coverage.

Reply:
@Patater @gilles-peskine-arm Please consider the new commit, I have prototyped the required change and tested it against a dummy `RSA_ALT` which forbids some private key derivations.